ui/src/app/components/with-new-user-satisfaction-survey-modal.spec.tsx

```diff
@@ -52,9 +53,9 @@ describe(withNewUserSatisfactionSurveyModal.name, () => {
     const validationMock = jest
       .spyOn(surveysApi(), 'validateOneTimeCodeForNewUserSatisfactionSurvey')
       .mockImplementationOnce(() => Promise.resolve(true));
-    const { queryByText } = await createWrapperAtPath(`?surveyCode=${code}`);
+    await createWrapperAtPath(`?surveyCode=${code}`);
     expect(validationMock).toHaveBeenCalledWith(code);
-    expect(queryByText(overallSatisfaction)).toBeInTheDocument();
+    expect(await screen.findByText(overallSatisfaction)).toBeInTheDocument();
```

Comment:
optional: could use the `getBy` versions (and no await) since we're expecting a match

Reply:
In most cases, I would agree that this would be preferred. However, in this case, the statement is the first assertion after rendering the component. The `findBy` has an implicit `waitFor` in it, so it waits for the overall satisfaction text to show up before passing/failing the assertion. It seems like RTL got a little stricter, in later versions, on how long it is willing to wait during non-async selectors. As written, this test will fail if `findBy` is swapped with `getBy`.

Reply:
oh right, I forgot about the "waitFor" in "findBy" - thanks.